wally-wally-wally/firmware
# Add BLDC Interface

**Purpose**
Create a python module to make it easier to control the BLDC motors

**Summary**
Code was developed to control the BLDC motors. It has functions to set the frequency of the PWM signal as well as control the direction and speed of the motor. All errors are based on the datasheet (frequency has to be in a certain range, speed cannot have a duty cycle less than 50%)

**Testing Done**
- Tried running the PWM signal with different frequencies (including testing out of bound frequencies to ensure that it throws the appropriate error)
- Tried both possible direction combinations (CW and CCW) as well as incorrect inputs to ensure that an error occurs
- Tried different speeds including the case that would throw an error

**Testing To Do**
- Ensure that it works with the main code
- Change CW/CCW to FORWARD/BACKWARD (if it makes it easier for us to understand - TBD, it would be based on how the motors are place in Wally and what is forward to one motor might be backwards to another so it might not work)
